### PR TITLE
Heatmap

### PR DESCRIPTION
**Pull request by andypasso**

### CODE REVIEW

**Comment by jofer86** [comment]
## Under review by Jorge F. (TSE)

File: index.html [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+  <title>Document</title>
+  <link rel="stylesheet" href="normalize.css" />
+  <link rel="stylesheet" href="style.css" />
+</head>
+
+<body>
+  <header>
```
Comment (jofer86) [inline index.html:14]: While this is semantically correct, we need to include, inside the `<header>` tag a `<nav>` tag in order to make it more appropriate, or just use a `<nav>` tag altogether, your choice, also it is recommended by the Miicroverse guidelines that the content of the `<nav>` consists of seven `<div>` tags, divided across the entire width, which you did, also as a suggestion they should cover the entire height of the `<nav>` tag .

File: index.html [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+      <li>Navigation</li>
+      <li>Navigation</li>
+      <li>Navigation</li>
+    </ul>
+  </header>
+  <main>
+    <div class="dontmiss">
+      <h1>Don’t Miss These Latest Articles</h1>
+    </div>
+    <div class="mainarticles">
```
Comment (jofer86) [inline index.html]: To get things semantically correct should follow these guidelines:
* [ ] All content should be wrapped with `<main>` HTML tags
* [ ] Every child of main should be a `<section>` HTML tag pair
* [ ] Every child of section should be an `<article>` HTML tag pair
**So in this case in particular our class of main articles should be a `<section>` tag and every child of it should be an `<article>` tag**

Also: 
* [ ] All four articles should have a width of `50%`
* [ ] Each container should have an image in the top left corner with the title of an article under it

**Please do make the necessary adjustments on this**

File: index.html [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        <h1>Create to make you think: Meet our new printed magazine</h1>
+      </div>
+      <div>
+        <h1>Create to make you think: Meet our new printed magazine</h1>
+      </div>
+      <div>
+        <h1>Create to make you think: Meet our new printed magazine</h1>
+      </div>
+    </div>
+    <div class="articles">
```
Comment (jofer86) [inline index.html]: For this part the only change necessary is changing that `<div>` tag and replace it with a `<section>` tag, the rest is done fine.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+
+      <article>
+        <h2>Monthly Web Development</h2>
+        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ex corporis alias delectus aut aliquid repudiandae
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+    </div>
+    <div class="publicity">
```
Comment (jofer86) [inline index.html]: Here is the same observation for the semantic markup, use a `<section>` tag containing `<article>` tags within it, and please make sure that the largest article contains a `title` and a small `paragraph`.

File: index.html [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+      <div class="slack">
+        <h1>slack</h1>
+      </div>
+
+      <div class="smashing">
+        <h1>1,521 smashing members</h1>
+      </div>
+    </div>
+    <div class="articles">
```
Comment (jofer86) [inline index.html]: For this part the only change necessary is changing that <div> tag and replace it with a <section> tag, the rest is done fine.

File: index.html [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+
+      <article>
+        <h2>Monthly Web Development</h2>
+        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ex corporis alias delectus aut aliquid repudiandae
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+    </div>
+    <div class="publicity">
```
Comment (jofer86) [inline index.html]: Here, use the same semantic markup said before, the wider element is positioned to the left here :+1: , but it does require a title and a small paragraph though.
**Please address those changes**

File: index.html [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+      </div>
+
+      <article>
+        <h2>Monthly Web Development</h2>
+        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ex corporis alias delectus aut aliquid repudiandae
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+
+    </div>
+    <div class="seemorearticles">
```
Comment (jofer86) [inline index.html]: Do use semantic markup here to please, should be a `<section>` tag.

File: index.html [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+      <h3> See more articles </h3>
+    </div>
+
+    <div class="comingup">
```
Comment (jofer86) [inline index.html]: Just the use of a semantically appropriate markup is necessary here section and article tags are necessary. The rest is fine.

File: index.html [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+      <h1>Smashing print : # ethics & stuff</h1>
+    </div>
+    <div class="personofweek">
+      <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatibus ut pariatur ipsam quisquam magni in ullam
+        illo nihil dolorum officiis. Asperiores adipisci quod voluptate, sit explicabo ut reiciendis ducimus? Earum?
+        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Debitis nam vero itaque molestiae tempora asperiores
+        delectus, repudiandae quam! Perferendis, vitae assumenda. Eum dolore accusantium molestiae et, inventore
+        doloremque odio commodi.</p>
+    </div>
+    <div class="articles2">
```
Comment (jofer86) [inline index.html]: `<section>` ` tags are necessary here too, the rest is done fine.

File: index.html [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+    </div>
+    <div class="seemorearticles">
+      <h3> See more articles </h3>
+    </div>
+
+    <div class="comingup">
+      <h1>Smashing print : # ethics & stuff</h1>
+    </div>
+    <div class="personofweek">
```
Comment (jofer86) [inline index.html]: Semantic markup necessary here too.

File: index.html [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+
+      <article>
+        <h2>Monthly Web Development</h2>
+        <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ex corporis alias delectus aut aliquid repudiandae
+          quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+      </article>
+    </div>
+    <div class="seemorearticles">
```
Comment (jofer86) [inline index.html]: Semantic markup should be placed in here too.

File: index.html [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+      <h3> See more on twitter </h3>
+    </div>
+    <div class="featuresmashingbook">
```
Comment (jofer86) [inline index.html]: We need some semantic markup here. Please address the changes.

File: index.html [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+      <div class="textwrap">
+        <h1> Design system</h1>
+        <p>Lorem ipsum, dolor sit amet consectetur adipisicing elit. Et enim animi, nobis quis esse nesciunt excepturi
+          nam
+          velit. Perspiciatis laborum aliquam dolorem eveniet eaque atque dolorum amet sequi debitis exercitationem!</p>
+      </div>
+    </div>
+    <div class="articles">
```
Comment (jofer86) [inline index.html]: Here we just need to change that `<div>` with a `<section>` tag and we are good.

File: index.html [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+            quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+        </article>
+  
+        <article>
+          <h2>Monthly Web Development</h2>
+          <p>Lorem ipsum dolor sit, amet consectetur adipisicing elit. Ex corporis alias delectus aut aliquid repudiandae
+            quaerat ad hic magnam pariatur quasi accusamus eaque provident, rerum officia sequi, cumque autem iusto!</p>
+        </article>
+      </div>
+    <div class="publicity">
```
Comment (jofer86) [inline index.html]: Semantic markup required here too

File: index.html [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+      <div class="newsletter smashing "><h1>Smashing Newsletter </h1> </div>
+
+      <div></div>
+    </div>
+    <div class="seemorearticles">
```
Comment (jofer86) [inline index.html]: Semantic markup required here too

File: index.html [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+        <h3> See more articles </h3>
+      </div>
+  </main>
+  <footer>
```
Comment (jofer86) [inline index.html:224]: Your footer should have an `<ul>` tag that contains `<li>` tags positioned with flex or floats, which I see you did but with t he `<a>` tags, please do adjust.

**Review by jofer86 — COMMENTED** [review]
The site was good, readable and understandable from the markup point of view, I did some observations on the code, most of them are in the lines of keeping these at all times in every section: 

**All content should be wrapped with `<main>` HTML tags
Every child of main should be a `<section>` HTML tag pair
Every child of a section should be an `<article>` HTML tag pair**

### You did a good job on the heat map part of the site,  but I need to make an observation there, remember that as we are scrolling down, the visual hierarchy might be resetting, mainly because we are not seeing the entirety of the site in one go. I would suggest to take this into consideration and do some re-arranging on the coloring. 

## Other that those minor changes I required was a great effort, please do tend to those changes and request a new code review.

# Cheers, Jorge F. :wave: :wave:

**Comment by samgaco** [comment]
### Hi @andynarf  and @AndresFMoya :vulcan_salute: 

### The project looks beautiful so far :+1: and I see you have worked through the suggested changes made by the previous reviewer. There are a few more points to address before proceeding:

* The **horizontal scroll bar** shouldn't be showing.

* This section **should have images in the top left corner** as you did with the first four half-width containers right after the navigation bar. Notice that **there is also a space** between the white border of the boxes at the right and the right border of the parent container with a grey background. **That should not be there.**
topleft

* **Suggestion**: Try looking up the warnings in the index file given by https://validator.w3.org

### That's it. Please :pray: work on these changes and submit back for a code review when you are done. Regards! :muscle:

**Comment by Ispirett** [comment]
**Great job with the changes in this project folks @andynarf @AndresFMoya :1st_place_medal:    you merge  this project now. Good luck on you next project.**

**Pull request merged**